lib/adapters/REST/endpoints/upload.ts

```diff
@@ -1,13 +1,23 @@
 import type { AxiosInstance } from 'contentful-sdk-core'
 import { Stream } from 'stream'
-import { GetSpaceParams } from '../../../common-types'
+import { GetSpaceEnvironmentUploadParams } from '../../../common-types'
 import { getUploadHttpClient } from '../../../upload-http-client'
 import { RestEndpoint } from '../types'
 import * as raw from './raw'
 
+const getUploadUrlPath = (params: GetSpaceEnvironmentUploadParams) => {
+  const spacePath = `/spaces/${params.spaceId}/uploads`
+  const environmentPath = `/spaces/${params.spaceId}/environments/${params.environmentId}/uploads`
+  const path = params.environmentId ? environmentPath : spacePath
+
+  // Return path as is if uploadId is not provided, otherwise append it to the path
+  if (!params.uploadId) return path
+  return path + `/${params.uploadId}`
+}
+
 export const create: RestEndpoint<'Upload', 'create'> = (
   http: AxiosInstance,
-  params: GetSpaceParams,
+  params: GetSpaceEnvironmentUploadParams,
```

Comment:
Basically this should then be `GetSpaceEnvironmentParams` right?

Reply:
Yes you are right!